nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/test/groovy/org/apache/nifi/properties/sensitive/aws/kms/AWSKMSSensitivePropertyProviderIT.groovy

```diff
@@ -0,0 +1,253 @@
+        CreateAliasRequest aliasReq = new CreateAliasRequest().withAliasName(aliasName).withTargetKeyId(dekResult.getKeyId())
+        client.createAlias(aliasReq)
+
+        // re-read the dek so we have the arn
+        DescribeKeyRequest descRequest = new DescribeKeyRequest().withKeyId(dekResult.getKeyId())
+        DescribeKeyResult descResult = client.describeKey(descRequest)
+
+        knownGoodKeys = [
+                dekResult.getKeyId(),
+                descResult.keyMetadata.getArn(),
+                aliasName
+        ]
+    }
+
+    @Before
+    void setUp() throws Exception {
+    }
+
+    @After
+    void tearDown() throws Exception {
+    }
+
+    /**
+     * This method schedules the deletion of the CMK created during setup.  The delete will cascade to the DEK and DEK alias.
+     */
+    @AfterClass
+    static void tearDownOnce() {
+        if (knownGoodKeys.size() > 0) {
+            ScheduleKeyDeletionRequest req = new ScheduleKeyDeletionRequest().withKeyId(knownGoodKeys[0]).withPendingWindowInDays(7)
+            client.scheduleKeyDeletion(req)
+        }
+    }
+
+
+    /**
+     * This test shows that bad keys lead to exceptions, not invalid instances.
+     */
+    @Test
+    void testShouldThrowExceptionsWithBadKeys() throws Exception {
+        SensitivePropertyProvider propProvider
+        String msg
+
+        msg = shouldFail(SensitivePropertyProtectionException) {
+            propProvider = new AWSKMSSensitivePropertyProvider("")
+        }
+        assert msg =~ "The key cannot be empty"
+
+
+        assert propProvider == null
+
+        msg = shouldFail(com.amazonaws.services.kms.model.NotFoundException) {
+            propProvider = new AWSKMSSensitivePropertyProvider("bad key")
+            propProvider.protect("value")
+        }
+        assert msg =~ "Invalid keyId"
+    }
+
+    /**
+     * These tests show that the provider with known keys can round-trip protect + unprotect random, generated text.
+     */
+    @Test
+    void testShouldProtectAndUnprotectValues() throws Exception {
+        SensitivePropertyProvider propProvider
+        String plainText
+
+        knownGoodKeys.each { k ->
+            propProvider = new AWSKMSSensitivePropertyProvider(k)
+            assert propProvider != null
+
+            byte[] randBytes = new byte[1024]
+            new SecureRandom().nextBytes(randBytes)
+            plainText = randBytes.encodeBase64()
+
+            assert plainText != null
+            assert plainText != ""
+
+            assert plainText == propProvider.unprotect(propProvider.protect(plainText))
+        }
+    }
+
+    /**
+     * These tests show that the provider cannot encrypt empty values.
+     */
+    @Test
+    void testShouldHandleProtectEmptyValue() throws Exception {
+        SensitivePropertyProvider propProvider
+        final List<String> EMPTY_PLAINTEXTS = ["", "    ", null]
```

Comment:
Maybe add `\n` as a value.

Reply:
Good catch; added `\n`, `\t` and a few combinations.